### PR TITLE
Gtk4 port

### PR DESCRIPTION
**Pull request by jwahlstrand**
I think this is ready to be tried out. Tests pass for me.

### CODE REVIEW

**Comment by jwahlstrand** [comment]
Argh, sorry for triggering CI again -- I meant to add a [skip ci] to prevent that. I guess it's freezing because it's using the registered version of GtkObservables, which depends on Gtk (?)

I've made some progress on the port of GtkObservables -- scroll-zoom now works. The missing piece is simulating mouse events with keyboard modifiers, which we need for testing and precompiling. I may need to add GdkEvent support to Gtk4.

**Comment by timholy** [comment]
No worries. If there's any way that commit privileges here and in GtkObservables would help your work, just ask and it will be granted.

**Comment by jwahlstrand** [comment]
> If there's any way that commit privileges here and in GtkObservables would help your work, just ask and it will be granted.

I'll take you up on that for GtkObservables. I'd like to run the CI tests for all platforms. It's working well for me locally now.

File: src/ImageView.jl [hunk 1]
```diff
@@ -359,9 +357,9 @@ Compat.@constprop :none function imshow_gui(canvassize::Tuple{Int,Int},
     if !isempty(slicedata)
         players = [player(slicedata.signals[i], axisvalues(slicedata.axs[i])[1]; id=i) for i = 1:length(slicedata)]
         guidict["players"] = players
-        hbox = Box(:h)
+        hbox = GtkBox(:h)
         for p in players
-            push!(hbox, frame(p))
+            #push!(hbox, frame(p))  # FIXME: causes freeze in tests
```
Comment (jwahlstrand) [inline src/ImageView.jl:362]: Tests are passing now that GtkObservables has been updated, but I'm currently stuck on this code involving `player`. It freezes Julia. Once I figure this out I think this is ready for merging.

**Comment by jwahlstrand** [comment]
All tests now pass, including the ones that were commented out before. I am going through the README and trying everything to make sure things work as advertised. I'm seeing a few things I want to fix before this is merged.

**Comment by timholy** [comment]
I took a brief look and everything seems uncontroversial. You can merge whenever you feel comfortable. Many thanks, I've been moving my lab and its so kind for you to carry this forward!

**Comment by jwahlstrand** [comment]
Great, thanks! One pesky issue remains -- when you zoom in on the pointer (control-scroll), it doesn't stay centered on the pointer. It zooms in and out and stays in the same general region as the pointer, but it's not as nice as before, and I haven't figured out why. I will give it another try this weekend, but if I can't fix it I may just merge and we can figure it out later. I generally have more time on weekends for Julia BTW.

As someone who moved my lab a few years ago, I feel your pain!

**Comment by jwahlstrand** [comment]
I did fix the zooming bug I mentioned, which was in GtkObservables.

**Comment by jwahlstrand** [comment]
Sure, 0.12.0 I guess? I'll have time to do it tomorrow morning.

**Comment by ashwani-rathee** [comment]
Yep, 0.12.0 would work

**Pull request merged**